LFGBulletinBoard/RequestList.lua

```diff
@@ -477,18 +477,48 @@ function GBB.GetDungeons(msg,name)
 	local runrequired=false
 	local hasrun=false
 	local runDungeon=""
+
+	if GBB.DB.TagsZhtw then
+		parts = GBB.Split(msg)
+	end
 		
 	for ip, p in pairs(parts) do
 		if p=="run" or p=="runs" then
 			hasrun=true
 		end
 		
 		local x=GBB.tagList[p]
+		if GBB.DB.TagsZhtw then
+			x=nil
+			for key,v in pairs(GBB.tagList) do
+				if strfind(p,key) then
+					if v==GBB.TAGSEARCH then
+						isGood=true
+						if x==nil then
+							x=v
+						end
+					elseif v==GBB.TAGBAD then
+						x=v
+						break
```

Comment:
Can you please explain the need to have almost duplicate logic specifically for zhtw?

Reply:
In Chinese server, we seldom use "space" for separating words in sentences.
For example, [4][SomeOne]: 英雄熔爐來坦補
英雄=heroic
熔爐=The Blood Furnace (dungeon)
來=LFM (TAGSEARCH)
坦=tank
補=heal
With this example, there is only one world "英雄熔爐來坦補" with multiple tag.
That's why duplicate logic in "one word" but not in "one sentence."

Reply:
Here is my main concern on logic on lines 491-508, is that you are looping through GBB.tagList every time inside a for loop, which might have some performance issues ie fps drops since this is O(N * M) (Arguably still O(N)). Maybe we can see if this can be refactored a little

Reply:
Here's what I thought.
Split in Chinese is completely unnecessary. (but important in English and etc.)
Chinese can walk another path only use strfind and not using Split, SplitNoNb, and table looking up.
Without Split, the performance in Chinese could reduced to O(N).
```
if not GBB.DB.TagsZhtw then
	parts =GBB.SplitNoNb(msg)
	for ip, p in pairs(parts) do
		local x=GBB.tagList[p]
		... (rest of code) ...
	end
else
	for key,v in pairs(GBB.tagList) do
		if strfind(msg,key) then
			local x=v
			... (rest of code) ...
		end
	end
end  
```

Reply:
Like you said Split and SplitNoNb doesn't matter since it doesn't really do a whole much in Chinese. That said for the example you provided  "英雄熔爐來坦補", `Split `and `SplitNoNb ` both produce the same result hence my comment on no need for doing `parts =GBB.Split(msg)` since we already do `local parts =GBB.SplitNoNb(msg)` earlier. Also the split isn't much of the problem. Now that I think about it iterating over `GBB.tagList` is kind of counterbalanced by `parts` being of size 1 in Chinese. 

I think we can leave it as is, can just clean up by removing
```
		if GBB.DB.TagsZhtw then
			for key,v in pairs(GBB.HeroicKeywords) do
				if strfind(p, key) then
					isHeroic = true
				end
			end
		end
```

and 

```
	if GBB.DB.TagsZhtw then
		parts = GBB.Split(msg)
	end
```

Reply:
A. Split part.
Here's another example that `SplitNoNb` produced strange result:
```
[4][SomeOne]: H破碎來坦補
```
which `破碎` is The Shattered Halls.
`SplitNoNb` would break `碎`
![20210616](https://user-images.githubusercontent.com/85930001/122142893-e94add00-ce82-11eb-89dd-63c2cf2ec0ae.png)

Reply:
In `GBB.HeroicKeywords part`, Chinese use `strfind` to match pattern not using table lookup.

Reply:
If we decide let Chinese walk another logic path, we can leave both code as they are.
Is this a good option? Code may looks like below:
```
if not GBB.DB.TagsZhtw then
	parts =GBB.SplitNoNb(msg)
	for ip, p in pairs(parts) do
		local x=GBB.tagList[p]
		... (rest of code) ...
	end
else
	for key,v in pairs(GBB.tagList) do
		if strfind(msg,key) then
			local x=v
			... (rest of code) ...
		end
	end
end  
```
I'll make another PR, if you think it's a good way to do so.

Reply:
Yea I think thats the better way to go. I noticed heroics not being properly identified (in English when having Chinese selected) unless you remove the parts I mentioned. So I think separating the two might be a good idea. Please do a test with having multiple languages selected (like English, Chinese, and Custom)

Reply:
I thinks it's very hard to identify properly when both Chinese and non-Chinese selected.
Currently use `GBB.DB.TagsZhtw` to tell which path should walk. 

Instead of using `GBB.DB.TagsZhtw` maybe we should make another chkBox, like `GBB.DB.STRFIND.`
This option tells using Chinese `strfind` method instead of original table lookup.

In Chinese local, set this option to true by default, otherwise set it to false by default.
Players can later change this option manually if they play EN server with Chinese client.

Reply:
Don’t worry about the identification. I’ll probably change it in the future to only having 1 language selected

Reply:
I'll create a new PR then.